include/gunrock/graph/detail/build.hxx

```diff
@@ -8,111 +8,229 @@
  *
  */
 
-#include <gunrock/graph/conversions/convert.hxx>
 #include <gunrock/algorithms/algorithms.hxx>
+#include <gunrock/formats/formats.hxx>
 
 namespace gunrock {
 namespace graph {
 namespace build {
 namespace detail {
 
 template <memory_space_t space,
-          view_t build_views,
           typename edge_t,
           typename vertex_t,
           typename weight_t>
-auto builder(vertex_t const& r,
-             vertex_t const& c,
-             edge_t const& nnz,
-             vertex_t* row_indices,
-             vertex_t* column_indices,
-             edge_t* row_offsets,
-             edge_t* column_offsets,
-             weight_t* values) {
+auto builder(graph::graph_properties_t properties,
+             format::csr_t<space, vertex_t, edge_t, weight_t>& csr) {
   // Enable the types based on the different views required.
   // Enable CSR.
-  using csr_v_t =
-      std::conditional_t<has(build_views, view_t::csr),
-                         graph::graph_csr_t<vertex_t, edge_t, weight_t>,
-                         empty_csr_t>;
-
-  // Enable CSC.
-  using csc_v_t =
-      std::conditional_t<has(build_views, view_t::csc),
-                         graph::graph_csc_t<vertex_t, edge_t, weight_t>,
-                         empty_csc_t>;
-
-  // Enable COO.
-  using coo_v_t =
-      std::conditional_t<has(build_views, view_t::coo),
-                         graph::graph_coo_t<vertex_t, edge_t, weight_t>,
-                         empty_coo_t>;
+  using csr_v_t = graph::graph_csr_t<vertex_t, edge_t, weight_t>;
+  using coo_v_t = empty_coo_t;
+  using csc_v_t = empty_csc_t;
 
```

Comment:
If we're explicitly calling the `builder` with whatever views we need, do we need to set unused types to empty? The graph_t struct removes already them.

In the following line you don't specify them:
```cpp
using graph_type = graph::graph_t<space, vertex_t, edge_t, weight_t, csr_v_t>;
```

What is the idea behind this?

Reply:
This is updated. Not sure what I was thinking there.